core/src/main/java/org/elasticsearch/tasks/TaskResult.java

```diff
@@ -160,7 +160,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     public XContentBuilder innerToXContent(XContentBuilder builder, Params params) throws IOException {
         builder.field("completed", completed);
-        builder.field("task", task);
```

Comment:
I missed this, I wonder if the XContentBuilder#field method should be rather made XContentObject aware by checking if the input `isFragment` and accordingly adding start and end object only when needed